crates/filament/src/ir_passes/fun_assumptions.rs

```diff
@@ -4,9 +4,9 @@ use fil_ir::{self as ir, AddCtx, Ctx, ExprIdx, PropIdx};
 
 /// Generates default assumptions to the Filament program for assumptions using custom functions
 #[derive(Default)]
-pub struct Assume;
+pub struct FunAssumptions;
```

Comment:
Nit: I don't love the name. Can we use something like `InferAssumes` which implies some relationship between existing assumptions and created ones?